### ISSUE TITLE
Prism duplicated regions

### ISSUE THREAD

**Issue by Qwin** [PrismLibrary/Prism#349]
Hi all,

This might be a big change but I have had quite a fight with it in the past. Mainly how we handle popouts with duplicate region. 

Example : 
I have 2 popouts with each its own region. Now within those regions I have subregions. And those subregions reuse a view which has a region (this is now duplicate on both popouts). The way prism solve this issue is by using scoped region managers.

However scoped region managers are horrible, a couple of things that is an issue for me :
-  new region managers are local and you would need something to keep track of them
- You cannot use dependency injection on the constructor because that one will get you only the root region manager
- Creating a scoped RegionManager is only possible on view level by creating your own region manager and then setting it on a region. Or using the parameter when registering a view to say that you want it scoped (a boolean just set it to true) and it will create a region manager for you and give that back as result. (now the last one you would need to create your own region beforehand and usually you would have your region in the view not in code)
- You need a place to do all this on the fly so usually you would set it up in the view / part of it in the viewmodel. I am now using the NavigatedTo to navigaterequest new reigons. 

What I would like to see in the future : 
- A better way of managing mutiple region without creating new region managers or at least to get a list of region managers with unique ids.
- A place for each view to register / navigate to other views (at the moment there isnt a really good place to place them)
- Scoped region manager creating when registering a view is just horrible why would you want that if you need to create a region in code and not in the xaml. 

Long story short if we can rethink the way we are handling scoped region managers that would be great. Thanks! I know this is quite some change but all I am saying is to keep this in mind for maybe future releases. 

Thanks,
Robert

**Comment by brianlagunas** [comment]
Your solution to managing scoped regions is here:

https://app.pluralsight.com/library/courses/prism-showing-multiple-shells/table-of-contents

and here:

https://app.pluralsight.com/library/courses/prism-mastering-tabcontrol/table-of-contents

The good news is that all your issues have a solution.

**Comment by Qwin** [comment]
Ok thanks for the links will check it out. :)

**Comment by Qwin** [comment]
I went through your videos and its a great solution, thank you for that! I couldn't find this solution online though (as you mentioned in the video). a "How To" section would be great indeed.

**Comment by brianlagunas** [comment]
Thanks for watching.  Those courses help pay to support Prism :)
